prefect_kubernetes/exceptions.py

```diff
@@ -0,0 +1,15 @@
+"""Module to define common exceptions within `prefect_kubernetes`."""
+
+from kubernetes.client.exceptions import ApiException
+
+
+class KubernetesJobDefinitionError(Exception):
+    """An exception for when a Kubernetes job definition is invalid"""
+
+
+class KubernetesJobFailedError(Exception):
+    """An exception for when a Kubernetes job fails"""
+
+
+class KubernetesResourceNotFoundError(ApiException):
+    """An exception for when a Kubernetes resource cannot be found by a client"""
```

Comment:
Can we define a base exception like KubernetesError and have these inherit from it so users can choose to catch more granular / broad exceptions?

Reply:
Or maybe inherit an exception from kubernetes.client.exceptions